src/implementations/twig/components/inpage-navigation/inpage-navigation.story.js

```diff
@@ -1,112 +1,127 @@
-  btnIdParagraph.innerHTML = demoText;
-  btnIdElement.insertAdjacentHTML('afterend', btnIdParagraph.outerHTML);
-  btnIdElement.insertAdjacentHTML('afterend', btnIdTag.outerHTML);
-  // Prevent the story from being reloaded.
+
+// Add generic container in the left or main column.
+const blockHandler = (region, state) => {
+  if (state) {
+    return `<div class="ecl-u-bg-grey-25 ecl-u-mb-m ecl-u-pa-l">
+              <h4 class="ecl-u-type-heading-4">This is an injected block in the ${region} column</h4>
+            </div>`;
+  }
+
   return false;
 };
 
-export default {
-  title: 'Components/Navigation/Inpage navigation',
+// Prepare data for the navigation.
+const prepareData = (data, args) => {
+  data.icon_path = defaultSprite;
+  data.links.forEach((item, i) => {
+    item.label = args[`heading${i + 1}`];
+  });
+
+  return data;
 };
 
-export const Default = () => {
-  // Buttons for the demo.
-  button(btnLeftLabel, btnLeftHandler, tabLabels.cases);
-  button(btnMainLabel, btnMainHandler, tabLabels.cases);
-  button(btnIdLabel, btnIdHandler, tabLabels.cases);
-  button(btnIdRemoveLabel, btnIdRemoveHandler, tabLabels.cases);
+// Prepare Html for the main content.
+const prepareHtmlContent = (data, args) => {
+  let html = '';
+  const lorem = loremIpsum({ count: 25 });
 
-  let pageFillerHtml = '';
-  demoData.links.forEach((content) => {
-    pageFillerHtml += content.item;
+  data.links.forEach((item, i) => {
+    const index = i + 1;
+    html += `<h2 class="ecl-u-type-heading-2" id="inline-nav-${index}">${
+      args[`heading${index}`]
+    }</h2>`;
+    html += `<p class="ecl-u-type-paragraph-m">${lorem}</p>`;
+    html += `<p class="ecl-u-type-paragraph-m">${lorem}</p>`;
   });
 
-  const fullDemoData = { ...demoData, icon_path: iconPath };
-  const html = inpageNavigation(fullDemoData);
-  const demo = document.createDocumentFragment();
-  const htmlElement = document.createElement('div');
-  htmlElement.innerHTML = `<div class="ecl-container">
-                            <div class="ecl-row ecl-u-mt-l" data-ecl-inpage-navigation-container>
-                              <div class="ecl-col-l-3">
-                                <div class="inPageDemoSidebar"></div>
-                                ${html}
-                              </div>
-                              <div class="ecl-col-l-9">
-                                <div class="inPageDemoContent"></div>
-                                ${pageFillerHtml}
-                              </div>
-                            </div>
-                          </div>`;
-  demo.appendChild(htmlElement.firstChild);
+  return html;
+};
+
+export default {
+  title: 'Components/Navigation/Inpage navigation',
+  parameters: {
+    knobs: {
+      disable: true,
+    },
+  },
+};
 
+export const Default = (args) => {
+  const navHtml = inpageNavigation(prepareData(demoData, args));
+  const contentHtml = prepareHtmlContent(demoData, args);
```

Comment:
I was initially confused about the reason why we were passing args to this function, but i see that also previously the change of the label in the knob was also producing a change in the heading in the content, i find it confusing, there is no reason why it should happen, the link between the two is the anchor value, the label of the in-page link is independent.

But then i also noticed that when migrating the component the specs were enriched by some content not in place in v2, like the lorem ipsum text, which we are not using here, maybe we could.

Reply:
I also wondered about this point. However, this is currently the case on v2, by changing the label of an element, it's updated in the nav and the content, I personally find that it makes more sense than just updating the navigation labels.

Reply:
I agree it's a bit confusing to have binding of args to data in this way. I tried locally and it seems to me demo data is good enough as Davide suggests